File: app/ui/dialogs/settings_dialog.py
from PySide6.QtWidgets import (QDialog, QVBoxLayout, QHBoxLayout, QLabel,
                             QComboBox, QPushButton, QGroupBox)
from app.utils.settings import settings

class SettingsDialog(QDialog):
    """Dialog window for configuring application preferences."""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setWindowTitle("Settings")
        self.setFixedSize(420, 360)

        layout = QVBoxLayout(self)
        layout.setContentsMargins(16, 16, 16, 16)
        layout.setSpacing(14)

        # AI Configuration
        grp_ai = QGroupBox("AI Background Removal Settings", self)
        vbox_ai = QVBoxLayout(grp_ai)

        hbox_model = QHBoxLayout()
        hbox_model.addWidget(QLabel("Default Model:", self))
        self.combo_model = QComboBox(self)
        self.combo_model.addItems(["RMBG-1.4", "U2Net", "Silueta"])
        self.combo_model.setCurrentText(settings.get("ai_model"))
        hbox_model.addWidget(self.combo_model)
        vbox_ai.addLayout(hbox_model)

        hbox_device = QHBoxLayout()
        hbox_device.addWidget(QLabel("Inference Hardware:", self))
        self.combo_device = QComboBox(self)
        self.combo_device.addItems(["Auto", "CUDA", "CPU"])
        self.combo_device.setCurrentText(settings.get("ai_device"))
        hbox_device.addWidget(self.combo_device)
        vbox_ai.addLayout(hbox_device)

        layout.addWidget(grp_ai)

        # GUI Theme Configuration
        grp_gui = QGroupBox("Interface & Appearance", self)
        vbox_gui = QVBoxLayout(grp_gui)

        hbox_theme = QHBoxLayout()
        hbox_theme.addWidget(QLabel("Theme:", self))
        self.combo_theme = QComboBox(self)
        self.combo_theme.addItems(["Dark", "Light"])
        self.combo_theme.setCurrentText(settings.get("theme"))
        hbox_theme.addWidget(self.combo_theme)
        vbox_gui.addLayout(hbox_theme)

        layout.addWidget(grp_gui)

        # Save / Cancel Buttons
        hbox_btns = QHBoxLayout()
        hbox_btns.addStretch()

        btn_cancel = QPushButton("Cancel", self)
        btn_cancel.clicked.connect(self.reject)
        hbox_btns.addWidget(btn_cancel)

        btn_save = QPushButton("Save Settings", self)
        btn_save.setObjectName("btn_primary")
        btn_save.clicked.connect(self._save_settings)
        hbox_btns.addWidget(btn_save)

        layout.addLayout(hbox_btns)

    def _save_settings(self):
        settings.set("ai_model", self.combo_model.currentText())
        settings.set("ai_device", self.combo_device.currentText())
        settings.set("theme", self.combo_theme.currentText())
        self.accept()

File: src/ui/dialogs/SettingsDialog.cpp
#include "ui/dialogs/SettingsDialog.h"
#include "utils/Settings.h"
#include <QVBoxLayout>
#include <QHBoxLayout>
#include <QGroupBox>
#include <QLabel>

namespace ImageCut {
namespace UI {

SettingsDialog::SettingsDialog(QWidget* parent)
    : QDialog(parent)
{
    setWindowTitle("Settings");
    setFixedSize(420, 360);

    QVBoxLayout* layout = new QVBoxLayout(this);
    layout->setContentsMargins(16, 16, 16, 16);
    layout->setSpacing(14);

    QGroupBox* grpAi = new QGroupBox("AI Background Removal Settings", this);
    QVBoxLayout* vboxAi = new QVBoxLayout(grpAi);

    QHBoxLayout* hboxModel = new QHBoxLayout();
    hboxModel->addWidget(new QLabel("Default Model:", this));
    m_comboModel = new QComboBox(this);
    m_comboModel->addItems({ "RMBG-1.4", "U2Net", "Silueta" });
    m_comboModel->setCurrentText(Utils::Settings::getInstance().get("ai_model", "RMBG-1.4"));
    hboxModel->addWidget(m_comboModel);
    vboxAi->addLayout(hboxModel);

    QHBoxLayout* hboxDevice = new QHBoxLayout();
    hboxDevice->addWidget(new QLabel("Inference Hardware:", this));
    m_comboDevice = new QComboBox(this);
    m_comboDevice->addItems({ "Auto", "CUDA", "CPU" });
    m_comboDevice->setCurrentText(Utils::Settings::getInstance().get("ai_device", "Auto"));
    hboxDevice->addWidget(m_comboDevice);
    vboxAi->addLayout(hboxDevice);

    layout->addWidget(grpAi);

    QGroupBox* grpGui = new QGroupBox("Interface & Appearance", this);
    QVBoxLayout* vboxGui = new QVBoxLayout(grpGui);

    QHBoxLayout* hboxTheme = new QHBoxLayout();
    hboxTheme->addWidget(new QLabel("Theme:", this));
    m_comboTheme = new QComboBox(this);
    m_comboTheme->addItems({ "Dark", "Light" });
    m_comboTheme->setCurrentText(Utils::Settings::getInstance().get("theme", "Dark"));
    hboxTheme->addWidget(m_comboTheme);
    vboxGui->addLayout(hboxTheme);

    layout->addWidget(grpGui);

    QHBoxLayout* hboxBtns = new QHBoxLayout();
    hboxBtns->addStretch();

    QPushButton* btnCancel = new QPushButton("Cancel", this);
    connect(btnCancel, &QPushButton::clicked, this, &QDialog::reject);
    hboxBtns->addWidget(btnCancel);

    QPushButton* btnSave = new QPushButton("Save Settings", this);
    btnSave->setObjectName("btn_primary");
    connect(btnSave, &QPushButton::clicked, this, &SettingsDialog::saveSettings);
    hboxBtns->addWidget(btnSave);

    layout->addLayout(hboxBtns);
}

void SettingsDialog::saveSettings() {
    Utils::Settings::getInstance().set("ai_model", m_comboModel->currentText());
    Utils::Settings::getInstance().set("ai_device", m_comboDevice->currentText());
    Utils::Settings::getInstance().set("theme", m_comboTheme->currentText());
    accept();
}

} // namespace UI
} // namespace ImageCut
